### PR TITLE
Initial support for cJSON

### PR DESCRIPTION
**Pull request by mywave82**
This is needed for updated version of Open Cubic Player

### CODE REVIEW

**Comment by mywave82** [comment]
All comments are welcome, this is the first package I am making

**Comment by augiedoggie** [comment]
According to portage the recipe should be in 'dev-libs/cjson' instead of 'dev-libs/libcjson'

File: dev-libs/libcjson/libcjson-1.7.15.recipe [hunk 1]
```diff
@@ -0,0 +1,72 @@
+SUMMARY="Ultralightweight JSON parser in ANSI C"
+DESCRIPTION="cJSON is an ultralightweight json parser."
+HOMEPAGE="https://github.com/DaveGamble/cJSON"
+COPYRIGHT="2009-2017, Dave Gamble and cJSON contributors"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://github.com/DaveGamble/cJSON/archive/refs/tags/v1.7.15.tar.gz"
```
Comment (korli) [inline dev-libs/libcjson/libcjson-1.7.15.recipe:7]:
```suggestion
SOURCE_URI="https://github.com/DaveGamble/cJSON/archive/v$portVersion.tar.gz"
```

File: dev-libs/libcjson/libcjson-1.7.15.recipe [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
```
Comment (korli) [inline dev-libs/libcjson/libcjson-1.7.15.recipe:8]:
```suggestion
```

**Comment by mywave82** [comment]
I will update the PR a bit later, thank you for all the feedback so far

**Comment by korli** [comment]
Looks good. Could you also enable "-DENABLE_CJSON_UTILS=On" and adjust the recipe?

**Comment by mywave82** [comment]
> Looks good. Could you also enable "-DENABLE_CJSON_UTILS=On" and adjust the recipe?

I have now added DENABLE_CJSON_UTILS=ON and flatten the commits to a single commit - reduces the noise in the commit log

**Pull request merged**